userspace/libsinsp/sinsp.cpp

```diff
@@ -131,6 +131,10 @@ sinsp::sinsp(bool with_metrics) :
 	// create state tables registry
 	m_table_registry = std::make_shared<libsinsp::state::table_registry>();
 	m_table_registry->add_table(m_thread_manager.get());
+
+#if !defined(__EMSCRIPTEN__)
+	m_thread_pool = std::make_shared<bs_thread_pool>();
```

Comment:
Honestly, I would make this `nullptr` by default, and then have every consumer decide whether or not the feature is enabled or not by having proper `set_thread_pool/get_thread_pool` accessors. For safety, this way you can also guard that the thread pool is not changed once first set in an inspector. This would protect you from cases where the threadpool can be changed between different plugins being registered (which would be a bit of a mess). Additionally, this would allow adopters to not even pull in the new library dependency unless strictly needed.

Reply:
I like not having to create a thread pool if nobody wants it, but do we have a suitable vtable around the thread pool? We can't take a pointer from a plugin and yolo cast it to `bs_thread_pool*`

Reply:
I think we don't cast to `bs_thread_pool*`, but to `thread_pool*` which is a class all with pure virtual methods. I think this should work as is, am I wrong?